### PR TITLE
Add tests to application

### PR DESCRIPTION
**Pull request by britaumich**
Since every new feature should have tests I would like to merge this branch first with rspec configuration and simple system tests.
After I added faker gem I run bundle and got this message: "Bundler could not find compatible versions for gem "i18n":"
I run "bundle update"

Then bundle gave me this message ""Unresolved or ambiguous specs during Gem::Specification.reset:
      diff-lcs (>= 1.2.0, < 2.0)"
I run "gem cleanup diff-lcs"

After that I did not have any errors.

I deleted files and folders from spec directory that we don't need.

System tests might require updating a chromedriver to match the localhost Chrome's version. The chromedriver can be downloaded from here https://chromedriver.chromium.org/downloads. Run "which chromedriver" to find a directory where to put a new chromedriver.

To run tests:
rspec spec/system/test-driver-spec.rb 
rspec spec/system/create_user-spec.rb 
rspec spec/system/home-page-spec.rb 
rspec spec/system/registration-spec.rb

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # This file is copied to spec/ when you run 'rails generate rspec:install'
 require 'spec_helper'
-ENV['RAILS_ENV'] ||= 'test'
+ENV['RAILS_ENV'] = 'test'
```
Comment (rsmoke) [inline spec/rails_helper.rb:3]: why get rid of the ||= for environment checks?

File: spec/system/create_user-spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
```
Comment (rsmoke) [inline spec/system/create_user-spec.rb:1]: file should be name `spec/system/create_user_spec.rb`

File: spec/system/home-page-spec.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
```
Comment (rsmoke) [inline spec/system/home-page-spec.rb:1]: file should with _spec.rb not -spec.rb

File: spec/system/registration-spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
```
Comment (rsmoke) [inline spec/system/registration-spec.rb:1]: file should with _spec.rb not -spec.rb

File: spec/system/test_seeds.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
+Gender.create!([
+  {name: "Female-1", description: "dudette"},
```
Comment (rsmoke) [inline spec/system/test_seeds.rb:2]: why `Female-1 ` and `Male-1`?

**Review by rsmoke — CHANGES_REQUESTED** [review]
in the gemfile on line 36 change `gem 'selenium-webdriver', '~> 3.142', '>= 3.142.7'` for 
  gem 'webdrivers' 
webdrivers has selenium as a dependency and get automatic installation and updates for all supported webdrivers

file `spec/system/create_user-spec.rb` should be named `spec/system/create_user_spec.rb`


add a file called `generators.rb` in config/initializers to exclude creation of areas that we are not interested in testing:

```
  Rails.application.config.generators do |g|
    g.test_framework :rspec,
      fixtures:         false,
      view_specs:       false,
      helper_specs:     false,
      routing_specs:    false,
      request_specs:    false,
      controller_specs: false
  end
```


lastly update the mysql2 gem to `gem 'mysql2', '~> 0.5.4'` to resolve the dependency issue. Note, you will need to run `gem install mysql2 -v '0.5.4' -- --with-ldflags=-L/usr/local/opt/openssl/lib --with-cppflags=-I/usr/local/opt/openssl/include`

File: spec/rails_helper.rb (hunk 1 above)
Comment (britaumich) [reply spec/rails_helper.rb:3]: Because if ENV['RAILS_ENV'] == 'DEVELOPMENT' it's not changing to 'TEST'

**Review by rsmoke — APPROVED** [review]

**Comment by rsmoke** [comment]
We will need to run the mysql2 0.5.4 on PROD when deploying.

**Pull request merged**